src/graph/LookupExecutor.cpp

```diff
@@ -0,0 +1,499 @@
+            if (!col->getFunName().empty()) {
+                return Status::SyntaxError("Not supported yet");
+            }
+            auto* prop = col->expr();
+            if (prop->kind() != Expression::kAliasProp) {
+                return Status::SyntaxError("Expressions other than AliasProp are not supported");
+            }
+            auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(prop);
+            if (schema->getFieldIndex(aExpr->prop()->c_str()) < 0) {
+                LOG(ERROR) << "Unknown column " << aExpr->prop()->c_str();
+                return Status::Error("Unknown column `%s' in schema", aExpr->prop()->c_str());
+            }
+            returnCols_.emplace_back(aExpr->prop()->c_str());
+        }
+    }
+    return Status::OK();
+}
+
+Status LookupExecutor::optimize() {
+    Status status = Status::OK();
+    do {
+        status = checkFilter();
+        if (!status.ok()) {
+            break;
+        }
+        status = chooseIndex();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    if (!status.ok()) {
+        LOG(ERROR) << "Preparing failed: " << status;
+        return status;
+    }
+    return status;
+}
+
+Status LookupExecutor::traversalExpr(const Expression *expr) {
+    switch (expr->kind()) {
+        case nebula::Expression::kLogical : {
+            auto* lExpr = dynamic_cast<const LogicalExpression*>(expr);
+            if (lExpr->op() == LogicalExpression::Operator::XOR) {
+                return Status::SyntaxError("Unsupported syntax ：%s", lExpr->toString().c_str());
+            } else if (lExpr->op() == LogicalExpression::Operator::OR) {
+                skipOptimize_ = true;
+            }
+            auto* left = lExpr->left();
+            traversalExpr(left);
+            auto* right = lExpr->right();
+            traversalExpr(right);
+            break;
+        }
+        case nebula::Expression::kRelational : {
+            std::string prop;
+            VariantType v;
+            auto* rExpr = dynamic_cast<const RelationalExpression*>(expr);
+            auto* left = rExpr->left();
+            auto* right = rExpr->right();
+            if (left->kind() == nebula::Expression::kAliasProp
+                && right->kind() == nebula::Expression::kPrimary) {
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(left);
+                prop = *aExpr->prop();
+                filters_.emplace_back(std::make_pair(prop, rExpr->op()));
+            } else if (left->kind() == nebula::Expression::kPrimary
+                       && right->kind() == nebula::Expression::kAliasProp) {
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(right);
+                prop = *aExpr->prop();
+                filters_.emplace_back(std::make_pair(prop, rExpr->op()));
+            } else {
+                return Status::SyntaxError("Unsupported syntax ：%s", rExpr->toString().c_str());
```

Comment:
It is not correct.  For example,  a > 1 && (b > 1 && c > 1) ,  it could use the index (a, b, c)

Reply:
The expr should be : 
LogicalExpr_root(RelationalExpr_1(a > 1)  AND LogicalExpr_1(RelationalExpr_2(b>1) AND RelationalExpr_3( c > 1))) . So this is a iteration and this index can be used.

Reply:
As your logic,  because the root expr is LogicalExpr(AND),  so it will fall in the second switch,  and the left expr is RelationExpr and right is LogicExpr,  so SyntaxError will be returned. right?

Reply:
> As your logic, because the root expr is LogicalExpr(AND), so it will fall in the second switch, and the left expr is RelationExpr and right is LogicExpr, so SyntaxError will be returned. right?

No ， This is an iterative function, if the root expr is LogicalExpr(AND), it should be call traversalExpr(left) and traversalExpr(right) again.